eclipse-theia/theia
# [preferences] use text models to update content

#### What it does


- fix #6845, fix #6786: use text models to update content. It resolves following issues:
  - edits are applied in the thread-safe fashion;
  - dirty editors are respected -> actually it is not good, it is better to way till a model is saved, otherwise intermediate values are applied.
    - [x] apply only saved values
- fix #7125, see the explanation: https://github.com/eclipse-theia/theia/issues/7125#issuecomment-584557345
- fix #7111, base language activation on monaco created modes
- activate textmate languages only if a grammar is registered, otherwise activation can be triggered too early and lead to no coloring at all
- don't swallow added file if it was deleted during debouncing, file resource rely on it to notify clients that is not valid anymore

#### How to test


- over 800 vscode compatibility preference tests are migrated to be integration tests
- (theming) test that changing themes does not lead to the conflict dialog anymore
- (vscode) use gitlens welcome screen to change different configuration and check that it does not lead to the conflict dialog anymore
- test 2 above steps with dirty editor on and off
- modify preferences and test that only saved preferences are applied
- check that formatting is respected when preference values are changed
- check that files with jsonc language are properly colored on first load and on reload with preserved editor

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
